src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -143,6 +159,7 @@ object RuleUtils {
             }
       }
       assert(filesByRelations.length == 1)
+      IndexLogEntryTags.resetHybridScanTagsIfNeeded(spark, plan, indexes)
```

Comment:
Do we have a test where configs change? If so, can you update the PR description?